src/praisonai/praisonai/scheduler/yaml_loader.py

```
@@ -134,24 +134,33 @@ def create_agent_from_config(agent_config: Dict[str, Any]) -> Any:
-                        tools.append(search_tool)
-                    except ImportError:
-                        logger.warning(f"Could not import {tool_name}, skipping")
-                # Add more tool imports as needed
+                resolver = _get_default_resolver()
+                # Reuse the same YAML shape resolve_all_from_yaml expects so a
+                # scheduled agents.yaml behaves like `praisonai run agents.yaml`.
+                resolved = resolver.resolve_all_from_yaml(
+                    {"roles": {name: agent_config}}
+                )
+                tools = list(resolved.values())
             except Exception as e:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)